### PR TITLE
death to maven! long live gradle!

### PR DESCRIPTION
**Pull request by thitu**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -28,3 +28,8 @@ deploy:
     script: scripts/deploy.sh
     on:
       tags: true
+
+notifications:
+  slack:
+    rooms:
+    - secure: "g80vE3QxtuRlgOlk8FTcmXN5xP8tzyKEb8fTLlW9Vs/GoiuTwIomqH5FjcGLSOsx/uXASqqH0AT3rvUD8lOClVxhFyK3MYdhcWHhKh6ocO2LyLfyZGMkhp1t3z+yUpp+pFiz6ci1BZ7HFvtdPnPKjO9HSBvXAQPGiotkqAtCBNs3o+ag7Xgkny2DKwUn1rvSEeaaWdpjzghnu/zW0L5yWpBT2R9Zw5CSG7BSeM1hRX4HzwZE/FjgwdFaEatQOWfhqC6nPXAp6JE/reJHzZpH49c2O1po0GZYwpOTVVvg0P/UfXge0miN5/VrtUheEWaeSEmgD4zrd0SLoP7Dnx9f+2m+UiHPegmb6+AWlwvfRFUgfQvZULFZvnGkLddRR/mhJchWvlTRHlENEI28fOBFnKV5Rnhyf3iziR4tZD1AGEsi1u+h+Ajv7CP4r40Q86jLKbSd2NwZNNfgi/qTZSeXhuSQ8ACFAtUFn4ENfAsbWD8H0NXnYpzIe1Dl0joqoHDE8xGRpuwqESKl+5/61gUv7bweqwOXIuzCXb2ti/oajGG85U1LrUYWKb3gCzk5d1cFxpP0g+hzNg5aq/acrK8qyl9ZiCpntVSqVX5c0GngyIqUI8MflF/ZR26FnnWd1qcLGxRf0Tx8IOtBH3tYTMBNC/kxIoqzcD6M9tp7IOv20iw="
```
Comment (alwold) [inline .travis.yml]: Is there any risk of having this in the public repo?

File: bin/semver-tool [hunk 2]
```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env bash
```
Comment (alwold) [inline bin/semver-tool]: Is this the same script that's in the common module? If we find a bug in it and need to fix the script, how do we keep them in sync?

File: build-extras.gradle [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  testLogging {
+    showStandardStreams = true
+    events "PASSED", "FAILED", "SKIPPED"
+  }
+
+  finalizedBy jacocoTestReport
+}
+
+configurations.all {
+  resolutionStrategy {
```
Comment (alwold) [inline build-extras.gradle]: What does this stuff do? Don't these just get pulled in automatically via the dependencies in build.gradle?

Comment (alwold) [reply build-extras.gradle]: did we actually run into that problem?

Comment (thitu) [reply build-extras.gradle]: yes - which is why i included the resolution strategy here. 
if we do not do this, there's no guarantee of which version of a dependency is picked at runtime.

**Pull request merged**